code/modules/mapping/reader.dm

```diff
@@ -55,7 +55,11 @@
 /datum/parsed_map/New(tfile, x_lower = -INFINITY, x_upper = INFINITY, y_lower = -INFINITY, y_upper=INFINITY, measureOnly=FALSE)
 	if(isfile(tfile))
 		original_path = "[tfile]"
+		var/temp_hack_garbage = tfile
 		tfile = rustg_file_read(tfile)
+		//This is hacky garbage
+		if(tfile == "") //Might be an upload, try raw loading it.
```

Comment:
Should use isfile

Reply:
Wait wha how does that first not work this is dodgy

Reply:
It reads file and checks if it's empty